opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanBuilder.java

```diff
@@ -24,36 +24,39 @@
  * by delegated builder internally. This is to avoid conditional check of different push down logic
  * for non-aggregate and aggregate query everywhere.
  */
-public class OpenSearchIndexScanBuilder extends TableScanBuilder {
+public abstract class OpenSearchIndexScanBuilder extends TableScanBuilder {
```

Comment:
follow up on same question on https://github.com/opensearch-project/sql/pull/1600.
why it is abstract and no subclass?